lib/http.c

```diff
@@ -1388,6 +1395,37 @@ static int http_getsock_do(struct connectdata *conn,
   return GETSOCK_WRITESOCK(0);
 }
 
+static void add_haproxy_protocol_header(struct connectdata *conn, bool *done)
+{
+  char proxy_header[128];
+  Curl_send_buffer *req_buffer;
+  CURLcode result;
```

Comment:
This compiler warning is of course because your function is 'void' and you never do anything with the return values in the function...
~~~
http.c:1404:12: error: variable ‘result’ set but not used [-Werror=unused-but-set-variable]
   CURLcode result;